File: nativelib/src/main/scala/scala/scalanative/native/errno.scala
package scala.scalanative
package native

@extern
object errno {
  @name("scalanative_errno")
  def errno: CInt = extern
  @name("scalanative_set_errno")
  def errno_=(value: CInt): Unit = extern
  @name("scalanative_edom")
  def EDOM: CInt = extern
  @name("scalanative_eilseq")
  def EILSEQ: CInt = extern
  @name("scalanative_erange")
  def ERANGE: CInt = extern
}

File: clib/src/main/resources/scala-native/errno.c
#include <errno.h>

int scalanative_errno() { return errno; }

void scalanative_set_errno(int value) { errno = value; }

int scalanative_edom() { return EDOM; }

int scalanative_eilseq() { return EILSEQ; }

int scalanative_erange() { return ERANGE; }
